sofria_mediaId.js

```diff
@@ -1,44 +1,144 @@
+
 const pk = new Proskomma();
 
-/*
- FIXME: find all files in specified directory which have extension .usx. The expectation is that each usx file contains one book
- Loop over each usx book file and associate with fqPath
- */
-
-// lines 20-33 will be in the book loop, process a single usx book file. Used later, book_id would be of the form 040MAT
-const fqPath = xx
-
-const content = fse.readFileSync(fqPath).toString();
-const suffix = fqPath.split(".").reverse()[0];
-/// FIXME: change xxx and yyy to empty strings
-pk.importDocument({lang: "xxx", abbr: "yyy"}, suffix, content);
-// I assume that xxx and yyy are embedded in the json somewhere?
-const chaptersQuery = `{documents {cIndexes { chapter } } }`;
-const chaptersResult = pk.gqlQuerySync(chaptersQuery);
-const ret = {};
-for (const chapter of chaptersResult.data.documents[0].cIndexes.map(ci => ci.chapter)) {
+const fqPath = process.argv[2]
+const usxRootDirectory = fqPath.split("/").reverse()[0];
+
+async function getListFileFromDirectory(dirPath) {
+    try {
+        const filesUsx = await fse.readdir(dirPath);
+        const listFiles = [];
+        for (const fileUsx of filesUsx) {
+            const fileFormat = fileUsx.split(".");
+            const name = fileFormat[0] ? fileFormat[0] : "";
+            const suffix = fileFormat[1] ? fileFormat[1] : "";
+
+            listFiles.push({
+                fullpath: path.join(dirPath, fileUsx),
+                name,
+                suffix,
+                fullname: fileUsx,
+            });
+        }
+        return listFiles;
+    } catch (err) {
+        console.error('Error to read directory!', path, err);
+    }
+}
+
+async function printChapterContent(usxFileName, chapter) {
     const chapterQuery = `{documents {sofria(indent: 2, chapter: ${chapter}) } }`;
-    // FIXME: instead of adding the chapter to a larger document to be returned, write one file per chapter
-    // example: given an input usx file ENGESVN_ET-usx/040MAT.usx, create ENGESVN_ET-json/040MAT_{chapter}.json
-    // where chapter is a zero-padded three-digit integer, one for each chapter in the input usx file.
-    // so just use the input filename head, with _{chapter}.json at the end
-    chapter_json = JSON.parse(pk.gqlQuerySync(chapterQuery).data.documents[0].sofria);
-    // output chapter_json to outfile
+    const gqlObject = pk.gqlQuerySync(chapterQuery);
+
+    if (gqlObject && gqlObject.data && gqlObject.data.documents[0]) {
+        const chapterJson = gqlObject.data.documents[0].sofria;
+        writeUsxJsonFile({
+            name: usxFileName,
+            chapter: chapter,
+            jsonContent: chapterJson,
+        });
+    }
+}
+
+async function generateJsonContentByUSXFile(usxFile) {
+    const content = fse.readFileSync(usxFile.fullpath).toString();
+    const stackChapterContent = [];
+
+    if (content !== "" && usxFile.suffix !== "") {
+        pk.importDocument({
+            lang: "xxx",
```

Comment:
change xxx and yyy to empty string

Reply:
@bradflood It has been replaced by "eng". E.g.

```javascript
pk.importDocument({
    lang: "eng",
    abbr: ""
}, usxFile.suffix, content);
```
The above because if we set `lang=""`, the Proskomma module will throw the following error:

`Error processing file listValue '' does not match regex '[a-z]{3}'`

Reply:
ok, I didn't realize that.
"eng" will be incorrect almost always. So please change it to xxx

Reply:
@bradflood , I have changed to `"xxx"`. See: https://github.com/faithcomesbyhearing/sofria-cli/pull/3/files#diff-26c5163d807f91f16662e922a42e39422ed5b39cb8bb6bb57d7260180a3dc273R75